cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -367,7 +357,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		scaleDown.CleanUp(currentTime)
 		potentiallyUnneeded := getPotentiallyUnneededNodes(autoscalingContext, allNodes)
 
-		typedErr := scaleDown.UpdateUnneededNodes(allNodes, potentiallyUnneeded, append(allScheduled, unschedulableWaitingForLowerPriorityPreemption...), currentTime, pdbs)
+		typedErr := scaleDown.UpdateUnneededNodes(allNodes, potentiallyUnneeded, originalScheduledPods, currentTime, pdbs)
```

Comment:
Shouldn't you pass scheduledPods here, not originalScheduledPods? That's how it worked before and it seems correct to me -> it makes sense for nominated pod to block scale-down of a node I think.

Reply:
hmm on the other hand there are pods to be preempted on that node already. Seems like using `scheduledPods` should be a bit safer approach. Yet both approaches seem kinda broken.